plugins/lookup/onepassword.py

```diff
@@ -95,31 +95,147 @@
     elements: str
 """
 
+import abc
 import errno
 import json
 import os
-
-from subprocess import Popen, PIPE
+import subprocess
 
 from ansible.plugins.lookup import LookupBase
 from ansible.errors import AnsibleLookupError
+from ansible.module_utils.common.process import get_bin_path
 from ansible.module_utils.common.text.converters import to_bytes, to_text
+from ansible.module_utils.six import with_metaclass
 
 from ansible_collections.community.general.plugins.module_utils.onepassword import OnePasswordConfig
 
 
+class OnePassCLIBase(with_metaclass(abc.ABCMeta, object)):
+    bin = "op"
+
+    def __init__(self, subdomain, domain, master_password, secret_key, vault):
+        self.subdomain = subdomain
+        self.domain = domain
+        self.master_password = master_password
+        self.secret_key = secret_key
+        self.vault = vault
+
+        self._path = None
+        self._version = None
+
+    @abc.abstractmethod
+    def get_account(self):
+        pass
+
+    @abc.abstractmethod
+    def get_item(self, item_id):
+        pass
+
+    @abc.abstractmethod
+    def signin(self):
+        pass
+
+    @abc.abstractmethod
+    def _parse_field(self, data_json, field_name, section_title):
+        pass
+
+    @property
+    def path(self):
+        if self._path is None:
+            self._path = get_bin_path(self.bin)
+
+        return self._path
+
+    @property
+    def version(self):
+        if self._version is None:
+            self._version = self.get_current_version()
+
+        return self._version
+
+    @classmethod
+    def get_current_version(cls):
+        """Standalone method to get the op CLI version. Useful when determing which class to load
+        based on the current version."""
+        try:
+            bin_path = get_bin_path(cls.bin)
+        except ValueError:
+            raise AnsibleLookupError("Unable to locate '%s' command line tool" % cls.bin)
+
+        try:
+            b_out = subprocess.check_output([bin_path, "--version"], stderr=subprocess.PIPE)
+        except subprocess.CalledProcessError as cpe:
+            raise AnsibleLookupError("Unable to get the op version: %s" % cpe)
+
+        return to_text(b_out).strip()
+
+    def _run(self, args, expected_rc=0, command_input=None, ignore_errors=False):
+        command = [self.path] + args
+        p = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE)
```

Comment:
You will want to replace `subprocess.Popen` with `module.run_command`
https://docs.ansible.com/ansible/latest/reference_appendices/module_utils.html

```bash
rc, stdout, stderr = module.run_command(command, data=command_input, encoding=None) 
```

Reply:
Since this is a lookup plugin and not a module, there is no `AnsibleModule` object, so `Popen` is fine.

Reply:
I was about to say that. 😄 As much as I'd love to use `run_command`, that's only possible for the module not either of the lookup plugins.